src/core/core.layouts.js

```diff
@@ -48,10 +94,141 @@ function findMaxPadding(boxes) {
 	};
 }
 
-function addSizeByPosition(boxes, size) {
-	helpers.each(boxes, function(box) {
-		size[box.position] += box.isHorizontal() ? box.height : box.width;
-	});
+function addSizeByPosition(outerDims, boxes) {
+	var result = extend({}, outerDims);
+	var i, ilen, layout, box;
+	for (i = 0, ilen = boxes.length; i < ilen; ++i) {
+		layout = boxes[i];
+		box = layout.box;
+		result[layout.pos] += layout.horizontal ? box.height : box.width;
+	}
+	return result;
+}
+
+function calculateMinSizes(boxes, params) {
```

Comment:
Thinking some more about this name, how about if we call it `fitToMinSizes` to make it clearer that it's mutating the input and align the name with `fitVerticalBoxes` / `fitHorizontalBoxes`

Reply:
has been removed since